### PR TITLE
Changes to add bulk api to public surface area

### PR DESCRIPTION
**Pull request by aayush3011**
1. Added bulk api to the public surface area.
2. Added Patch API to the public surface area.
3. Added Bulk and Patch API test cases.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java [hunk 1]
```diff
@@ -359,6 +363,60 @@ public CosmosBatchResponse executeCosmosBatch(
...
+     * <p>
+     *     We can get the corresponding operation using {@link CosmosBulkOperationResponse#getOperation()} and
+     *     it's response using {@link CosmosBulkOperationResponse#getResponse()}. If the operation was executed
+     *     successfully, the value returned by {@link com.azure.cosmos.models.CosmosBulkItemResponse#isSuccessStatusCode()} will be true. To get
+     *     actual status use {@link com.azure.cosmos.models.CosmosBulkItemResponse#getStatusCode()}.
+     * </p>
+     * To check if the operation had any exception, use {@link CosmosBulkOperationResponse#getException()} to
+     * get the exception.
+     */
+    public <TContext> Flux<CosmosBulkOperationResponse<TContext>> executeBulkOperations(
```
Comment (simplynaveen20) [inline sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java:386]: This is sync container we dont return flux or mono , please check other methods

Comment (aayush3011) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java]: Fixed the code.

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java [hunk 2]
```diff
@@ -359,6 +365,60 @@ public CosmosBatchResponse executeCosmosBatch(
         return this.blockBatchResponse(this.cosmosEncryptionAsyncContainer.executeCosmosBatch(cosmosBatch, requestOptions));
     }
 
+    /**
+     * Executes flux of operations in Bulk.
+     *
+     * @param <TContext> The context for the bulk processing.
+     * @param operations Flux of operation which will be executed by this container.
+     *
+     * @return A Flux of {@link CosmosBulkOperationResponse} which contains operation and it's response or exception.
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java:374]: Documentation is in-correct, please fix it here and in below API.

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java [hunk 3]
```diff
@@ -359,6 +365,60 @@ public CosmosBatchResponse executeCosmosBatch(
         return this.blockBatchResponse(this.cosmosEncryptionAsyncContainer.executeCosmosBatch(cosmosBatch, requestOptions));
     }
 
+    /**
+     * Executes flux of operations in Bulk.
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java:369]: There is no flux in sync container, please fix documentation for these APIs.

**Review by kushagraThapar — CHANGES_REQUESTED** [review]

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java (hunk 1 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java]: Are there missing tests for this?
If not, how can flux come in sync container, and still work fine?
@simplynaveen20 / @aayush3011  please make sure to add tests for these newly added APIs.

**Review by FabianMeiswinkel — CHANGES_REQUESTED** [review]
Please add test coverage

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java (hunk 2 above)
Comment (aayush3011) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java]: Fixed

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java (hunk 3 above)
Comment (aayush3011) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java]: Fixed

**Comment by aayush3011** [comment]
> Please add test coverage

Added test coverage

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java (hunk 1 above)
Comment (aayush3011) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java]: @kushagraThapar, I have added the tests as well now.

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Review by kushagraThapar — APPROVED** [review]

**Review by simplynaveen20 — APPROVED** [review]
LGTM , Thanks

**Pull request merged**